File: language/src/main/java/brj/nodes/VectorNode.java
package brj.nodes;

import brj.BridjeLanguage;
import brj.runtime.BridjeVector;
import com.oracle.truffle.api.dsl.Specialization;

public abstract class VectorNode extends CollNode {

    protected VectorNode(BridjeLanguage lang) {
        super(lang);
    }

    @Override
    @Specialization
    BridjeVector executeColl(Object[] res) {
        return new BridjeVector(res);
    }
}

File: language/src/main/kotlin/brj/nodes/VectorNode.kt
package brj.nodes

import brj.BridjeNode
import brj.runtime.BridjeVector
import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.nodes.Node.Child
import com.oracle.truffle.api.source.SourceSection

class VectorNode(
    @field:Child private var els: ExecuteCollNode,
    loc: SourceSection? = null
) : BridjeNode(loc) {

    override fun execute(frame: VirtualFrame): Any = BridjeVector(els.execute(frame))
}
